stage0/src/asm/boot.s

```diff
@@ -73,10 +73,59 @@ _protected_mode_start:
     # Set up a basic stack, as we may get interrupts.
     mov $stack_start, %esp
 
+    # Zero-out all the page tables: start address in EDI, value in AL, count in ECX.
+    mov $4906, %ecx           # each table is exactly 4096 bytes long.
+    xor %eax, %eax            # zero out eax
+    mov ${pml4}, %edi
+    rep stosb
+
+    mov ${pdpt}, %edi
+    rep stosb
+
+    mov ${pd_0}, %edi
+    rep stosb
+
+    mov ${pd_3}, %edi
+    rep stosb
+
+    mov ${pt_0}, %edi
+    rep stosb
+
+    # Set the first entry of PML4 to point to PDPT (0..512GiB).
+    mov ${pml4}, %eax
+    mov ${pdpt}, %edi
+    orl $3, %edi              # edi |= 3 (PRESENT and WRITABLE)
+    mov %edi, (%eax)          # set first half of PML4[0]
```

Comment:
Do you have to go through the indirection of having the address in EAX?

Reply:
I can where there is no offset involved, so updated those.